### PR TITLE
Fix cabal file, closes #6

### LINKED ISSUE #6: Add other modules to test suite
There's no other modules section in the test suite component, so the Hackage package has no tests:

```
Warning: modules not listed in atomic-write.cabal for 'atomic-write-test' component (add to other-modules):
    System.AtomicWrite.Internal
    System.AtomicWrite.Writer.ByteString
    System.AtomicWrite.Writer.ByteStringBuilder
    System.AtomicWrite.Writer.ByteStringBuilderSpec
    System.AtomicWrite.Writer.ByteStringSpec
    System.AtomicWrite.Writer.LazyByteString
    System.AtomicWrite.Writer.LazyByteStringSpec
    System.AtomicWrite.Writer.String
    System.AtomicWrite.Writer.StringSpec
    System.AtomicWrite.Writer.Text
    System.AtomicWrite.Writer.TextSpec
```

Also, the atomic-write library should be listed as a build dependency of the test suite.

### PR DESCRIPTION
**Pull request by acamino**
### Add other modules to test suite

- Add spec modules to the test suite.
- Add atomic-write library as a build dependency of the test suite.

Solves #6.

### CODE REVIEW

File: atomic-write.cabal [hunk 1]
```diff
@@ -72,10 +72,17 @@ library
...
 
+  other-modules:       System.AtomicWrite.Writer.ByteStringSpec
+                     , System.AtomicWrite.Writer.ByteStringBuilderSpec
+                     , System.AtomicWrite.Writer.LazyByteStringSpec
+                     , System.AtomicWrite.Writer.StringSpec
+                     , System.AtomicWrite.Writer.TextSpec
+
   build-depends:       base >= 4.5 && < 5.0
+                     , atomic-write
                      , temporary
```
Comment (sestrella) [inline atomic-write.cabal]: @acamino do you double checking if we need all this dependencies? Probably we include them to build all modules defined at `src`. But now that we include `atomic-write` as a dependency probably we don't need all these dependencies.

Comment (acamino) [inline atomic-write.cabal]: I double check them. Those modules are defined as specs. I'm not requiring the modules defined at src. 😉 
If I don't add those dependencies (other-modules) we'll end with a tarball like this https://hackage.haskell.org/package/atomic-write-0.2.0.5/src/ where the spec modules are excluded. What do you think?

Comment (acamino) [inline atomic-write.cabal]: I like the idea of failing if there is a _warning_. I will add the `-Werror` option as you suggested.

Comment (sestrella) [inline atomic-write.cabal]: @acamino about:

> I double check them. Those modules are defined as specs. I'm not requiring the modules defined at src. :wink:
If I don't add those dependencies (other-modules) we'll end with a tarball like this https://hackage.haskell.org/package/atomic-write-0.2.0.5/src/ where the spec modules are excluded. What do you think?

I was taking about the dependencies (e.g. time, temporary)  not the exported modules :)

Comment (acamino) [inline atomic-write.cabal]: Gotcha. Thanks @sestrella!

**Comment by sestrella** [comment]
@acamino LGTM

**Pull request merged**